File: javascript/easy/keyboardRow.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var findWords = function(words) {
    const rowInfo = [1,2,2,1,0,1,1,1,0,1,1,1,2,2,0,0,0,0,1,0,0,2,0,2,0,2];
    let res = [];
    for(let word of words) {
        let row = rowInfo[ord(word[0])], canBeTyped = true;
        for(let c of word) {
            if(rowInfo[ord(c)] !== row) {
                canBeTyped = false;
                break;
            }
        }
        if(canBeTyped) {
            res.push(word);
        }
    }
    return res;
};

var ord = function(c) {
    let ascii = c.charCodeAt(0);
    return (ascii >= 65 && ascii < 97) ? (ascii - 'A'.charCodeAt(0)) : (ascii - 'a'.charCodeAt(0));
};

File: ruby/easy/keyboard_row.rb
# @param {String[]} words
# @return {String[]}
def find_words(words)
    row_info = [1,2,2,1,0,1,1,1,0,1,1,1,2,2,0,0,0,0,1,0,0,2,0,2,0,2]
    res = []
    words.each do |word|
        row = row_info[ord(word[0])]
        can_be_typed = true
        word.each_char do |c|
            can_be_typed = false && break if row_info[ord(c)] != row
        end
        res.push(word) if can_be_typed
    end
    return res
end

def ord(c)
    return (c >= 'A' && c <= 'Z') ? (c.ord - 'A'.ord) : (c.ord - 'a'.ord)
end
